### PR TITLE
move some stuff to constants

### PR DESCRIPTION
**Pull request by jonathannewell**

### CODE REVIEW

**Comment by jonathannewell** [comment]
@trayo Any word on this? If this gets merged I can update my other pull request and hopefully we can tie this stuff off. Thoughts? Cheers, Jon

File: constants/argument_constants.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package constants
+
+const (
+	DEFAULT_CONFIG_FILE_PATH = "./config.yml"
```
Comment (trayo) [inline constants/argument_constants.go]: Can you explain the benefit of having a constant for these strings that are only used in server.go?

I think I'd rather see this stay and just have a constants file for the events.

File: constants/environment_constants.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package constants
+
+const (
+	LOG_LEVEL_ENV_VAR_NAME = "DEPLOYADACTYL_LOGLEVEL"
```
Comment (trayo) [inline constants/environment_constants.go]: Same as above.

File: controller/deployer/deployer.go [hunk 3]
```diff
@@ -132,7 +133,7 @@ func (d Deployer) Deploy(req *http.Request, environment, org, space, appName, co
 
 	e, found := environments[deploymentInfo.Environment]
 	if !found {
-		err = d.EventManager.Emit(S.Event{Type: "deploy.error", Data: deployEventData})
```
Comment (trayo) [inline controller/deployer/deployer.go]: These should also change in the `deployer_test.go` and any other test files they are being used in.

File: constants/argument_constants.go (hunk 1 above)
Comment (jonathannewell) [inline constants/argument_constants.go]: @trayo I can pull it out. Mostly that's just me. Not a fan of magic strings and I was just trying to get stuff in one place. Having the package contain one constants file it just seemed lonely. This is a constant back and forth for me. Keep things in one place for ease of maintenance or keep them with the code where they are used.  Either way if you want to put that stuff back I will.

File: controller/deployer/deployer.go (hunk 3 above)
Comment (jonathannewell) [inline controller/deployer/deployer.go]: Yeah, missed that!

File: constants/argument_constants.go (hunk 1 above)
Comment (trayo) [inline constants/argument_constants.go]: We used to have something similar for all of our error strings and they ended up causing headaches in the tests.

I think we can move forward with just the constants for events.

File: constants/event_constants.go [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package constants
+
+const (
+	DEPLOY_START_EVENT   = "deploy.start"
```
Comment (trayo) [inline constants/event_constants.go]: Sorry to be so nitpicky, but I just realized these should match the go standard and be camel case like `DeployStartEvent`.

File: server.go [hunk 5]
```diff
@@ -12,17 +12,20 @@ import (
 )
 
 const (
-	defaultConfig = "./config.yml"
-	defaultLevel  = "DEBUG"
+	default_config_file_path = "./config.yml"
+	config_file_arg          = "config"
```
Comment (trayo) [inline server.go]: Same as above this should be `configFileArg`.

**Comment by trayo** [comment]
Thanks!

**Comment by trayo** [comment]
@jonathannewell I want to mention that in order to get the commits tied to your github account you need to commit using the email you use to login. Our CLA check will show unsigned on the commits, but I've checked manually it's been done. Just wanted to make sure you know.

**Pull request merged**